src/arrayMethodReduce.js

```diff
@@ -5,7 +5,25 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    if (this.length === 0) {
+      return initialValue !== undefined ? initialValue : 'undefined';
+    }
+
+    let startIdx = 0;
+    let result;
+
+    if (arguments.length === 2) {
+      result = initialValue;
+    } else {
```

Comment:
Let's simplify it a little bit
```suggestion
    let result = initialValue;

    if (... (add the correct condition here)) {
```